### PR TITLE
NER support for HF datasets

### LINKED ISSUE #579: Add NER support for HF datasets
The format we are going to support will be like list of tokens and list of labels:

Image

Example: https://huggingface.co/datasets/germaner

### PR DESCRIPTION
**Pull request by alytarik**
Example Usage and Dataset:
image
image

closes  #579 

## Checklist:

- [x] I've added Google style docstrings to my code.
- [x] I've used `pydantic` for typing when/where necessary.
- [x] I have linted my code
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: langtest/datahandler/datasource.py [hunk 1]
```diff
@@ -1088,3 +1115,50 @@ def _row_to_sample_classification(self, data_row: Dict[str, str]) -> Sample:
...
+                NERPrediction.from_span(
+                    entity=label,
+                    word=token,
+                    start=cursor,
+                    end=cursor + len(token),
+                    doc_id=0,
+                    doc_name="",
+                    pos_tag="XX",
+                    chunk_tag="XX",
+                )
```
Comment (JulesBelveze) [inline langtest/datahandler/datasource.py:1156]: Just a random thought: we should probably get rid of `pos_tag` and `chunk_tag`, we not using them anywhere

Comment (JulesBelveze) [reply langtest/datahandler/datasource.py:1156]: But not relevant in this PR

File: langtest/langtest.py [hunk 2]
```diff
@@ -164,7 +165,19 @@ def __init__(
...
+            isinstance(data, dict)
+            and hub in self.SUPPORTED_HUBS_HF_DATASET_NER
+            and task == "ner"
+        ):
+            self.data = HuggingFaceDataset(data["name"], task=task).load_data(
+                feature_column=data.get("feature_column", "tokens"),
+                target_column=data.get("target_column", "ner_tags"),
+                split=data.get("split", "test"),
+                subset=data.get("subset", None),
+            )
```
Comment (JulesBelveze) [inline langtest/langtest.py:177]: What we could probably do here is to remove the default parameter values of the `load_data` method and only set them in the `load_data_ner`, `load_data_XXX` methods. That is gonna easier to extend to new tasks.

**Review by JulesBelveze — CHANGES_REQUESTED** [review]
Minor comments. Also, could you please add a few lines to the PR description explaining the goal of this PR (even though quite self explanatory).

Also, I think there's a problem with the way you parametrize the test, and that's why it fails.

File: langtest/datahandler/datasource.py [hunk 3]
```diff
@@ -914,9 +918,9 @@ def load_data_ner(
 
     def load_data_classification(
         self,
-        feature_column: str = "text",
-        target_column: str = "label",
-        split: str = "test",
+        feature_column: str = None,
+        target_column: str = None,
+        split: str = None,
```
Comment (JulesBelveze) [inline langtest/datahandler/datasource.py:923]: Please remove default values here

File: langtest/datahandler/datasource.py [hunk 4]
```diff
@@ -888,12 +888,16 @@ def _check_datasets_package(self):
 
     def load_data_ner(
         self,
-        split: str,
+        feature_column: str = None,
+        target_column: str = None,
+        split: str = None,
```
Comment (JulesBelveze) [inline langtest/datahandler/datasource.py:893]: Please remove default values here

File: tests/test_datasource.py [hunk 5]
```diff
@@ -169,7 +174,7 @@ def test_load_raw_data(self, dataset, feature_col, target_col):
     def test_load_data(self, dataset, feature_col, target_col):
         """"""
         if isinstance(dataset, HuggingFaceDataset):
-            samples = dataset.load_data(split="test[:30]")
+            samples = dataset.load_data(split="test")
         else:
```
Comment (JulesBelveze) [inline tests/test_datasource.py:178]: Any reason why you remove the `[:30]`?

Comment (alytarik) [reply tests/test_datasource.py:178]: I was trying to fix a broken test then just kept it.

**Review by JulesBelveze — APPROVED** [review]

**Pull request merged**